library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java

```diff
@@ -516,53 +528,78 @@ private int findFormatFromChannels(int numChannels) {
         }
     }
 
-    public void initStream() throws IOException {
-        mLock.lock();
+    private String currentPath() {
+        if (mPath != null) {
+            return mPath;
+        }
+        else if (mUri != null) {
+            return mUri.toString();
+        }
+
+        return null;
+    }
+
+    public boolean initStream() throws IOException {
         mExtractor = new MediaExtractor();
+
+        String lastPath = currentPath();
```

Comment:
Can you add a comment why this is necessary?
Might be useful, especially if someone tries to understand this in a few months...